services/src/main/java/org/fao/geonet/api/registries/vocabularies/KeywordsApi.java

```diff
@@ -308,7 +308,135 @@
         return searcher.getResults();
     }
 
-
+    /**
+     * Search keywords.
+     *
+     * @param q the q
+     * @param lang the lang
+     * @param rows the rows
+     * @param start the start
+     * @param targetLangs the target langs
+     * @param thesaurus the thesaurus
+     * @param type the type
+     * @param uri the uri
+     * @param sort the sort
+     * @param request the request
+     * @param httpSession the http session
+     * @return the list
+     * @throws Exception the exception
+     */
+    @ApiOperation(
+        value = "Search keywords (XML)",
+        nickname = "searchKeywordsXml",
+        notes = "")
+    @RequestMapping(
+        path = "/search",
```

Comment:
@josegar74 In swagger we noticed that if we have the same path in 2 different methods then only one method is displayed in the API doc right ? So here we have to only have one method on `/search`  which handles both MediaType JSON and XML. Correct ?

Reply:
@fxprunayre I believe you are correct. Options include
- Using different path - i.e. `/searchXml` and `/searchJson`
- Using different request method - i.e one being `get` and one being `post`
- Or as you suggested use one /search path and have it consume both xml and json

I believe your suggestion seems to be the better option in this case.

Reply:
> * Or as you suggested use one /search path and have it consume both xml and json

Added this option in https://github.com/fxprunayre/core-geonetwork/commit/9d031afa52f1e53adc7b829cd66e161dae972a64.  @alcinnz I let you double check and cherry-pick this commit in your branch if fine for you ? I also applied the change to ISO19139 (Lots of people are still using it). Then we can merge here. For next PR, you should target master first.